kani-compiler/src/codegen_aeneas_llbc/mir_to_ullbc/mod.rs

```diff
@@ -0,0 +1,799 @@
+        let body = self.translate_body(mir_body);
+        self.translated.bodies.set_slot(body_id, body);
+        Ok(body_id)
+    }
+
+    fn translate_body(&mut self, mir_body: Body) -> CharonBody {
+        let span = self.translate_span(mir_body.span);
+        let arg_count = self.instance.fn_abi().unwrap().args.len();
+        let locals = self.translate_body_locals(&mir_body);
+        let body: BodyContents =
+            mir_body.blocks.iter().map(|bb| self.translate_block(bb)).collect();
+
+        let body_expr = ExprBody { span, arg_count, locals, body };
+        CharonBody::Unstructured(body_expr)
+    }
+
+    fn requires_caller_location(&self, instance: Instance) -> bool {
+        let instance_internal = rustc_internal::internal(self.tcx(), instance);
+        instance_internal.def.requires_caller_location(self.tcx())
+    }
+
+    fn translate_ty(&self, ty: Ty) -> CharonTy {
+        match ty.kind() {
+            TyKind::RigidTy(rigid_ty) => self.translate_rigid_ty(rigid_ty),
+            _ => todo!(),
+        }
+    }
+
+    fn translate_rigid_ty(&self, rigid_ty: RigidTy) -> CharonTy {
+        debug!("translate_rigid_ty: {rigid_ty:?}");
+        match rigid_ty {
+            RigidTy::Bool => CharonTy::Literal(LiteralTy::Bool),
+            RigidTy::Char => CharonTy::Literal(LiteralTy::Char),
+            RigidTy::Int(it) => CharonTy::Literal(LiteralTy::Integer(translate_int_ty(it))),
+            RigidTy::Uint(uit) => CharonTy::Literal(LiteralTy::Integer(translate_uint_ty(uit))),
+            RigidTy::Never => CharonTy::Never,
+            RigidTy::Str => CharonTy::Adt(
+                TypeId::Builtin(BuiltinTy::Str),
+                // TODO: find out whether any of the information below should be
+                // populated for strings
+                GenericArgs {
+                    regions: Vector::new(),
+                    types: Vector::new(),
+                    const_generics: Vector::new(),
+                    trait_refs: Vector::new(),
+                },
+            ),
+            RigidTy::Ref(region, ty, mutability) => CharonTy::Ref(
+                self.translate_region(region),
+                Box::new(self.translate_ty(ty)),
+                match mutability {
+                    Mutability::Mut => RefKind::Mut,
+                    Mutability::Not => RefKind::Shared,
+                },
+            ),
+            RigidTy::Tuple(ty) => {
+                let types = ty.iter().map(|ty| self.translate_ty(*ty)).collect();
+                // TODO: find out if any of the information below is needed
+                let generic_args = GenericArgs {
+                    regions: Vector::new(),
+                    types,
+                    const_generics: Vector::new(),
+                    trait_refs: Vector::new(),
+                };
+                CharonTy::Adt(TypeId::Tuple, generic_args)
+            }
+            RigidTy::FnDef(def_id, _args) => {
```

Comment:
I don't think you should be ignoring `args` here. You need to instantiate the type even when handling generic code.

Reply:
I added an `unimplemented` for non-empty `args`. Would that work?